src/Avalonia.Base/AvaloniaPropertyRegistry.cs

```diff
@@ -47,7 +88,7 @@ public class AvaloniaPropertyRegistry
         public IReadOnlyList<AvaloniaProperty> GetRegistered(Type type)
         {
             _ = type ?? throw new ArgumentNullException(nameof(type));
-
+            _unregisteringManualResetEvent.WaitOne();
```

Comment:
I don't think it's really thread-safe?

Reply:
It's possible for the following TryGetValue to be executing when  `_unregisteringManualResetEvent.Reset();` is hit, so two threads would be changing the registry simultaneously.

We don't really support multithreading for AvaloniaObject yet, but it's still worth considering, since we'll want to have at least headless mode for non-UI threads rather soon.

Reply:
I used lock instead of ManualResetEvent, please review it again